cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,366 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
+    get_filename_component(PLUGIN_NAME "${PLUGIN_PATH}" NAME_WLE )
+    set(default_install lib)
+    if(ign_add_plugin_GUI)
+
+      QT5_WRAP_CPP(${PLUGIN_NAME}_headers_MOC ${path_to_plugins}/${PLUGIN_NAME}.hh)
+      QT5_ADD_RESOURCES(${PLUGIN_NAME}_RCC ${path_to_plugins}/${PLUGIN_NAME}.qrc)
+
+      set(PLUGIN_PATH ${PLUGIN_PATH}
+        ${${PLUGIN_NAME}_headers_MOC}
+        ${${PLUGIN_NAME}_RCC}
+      )
+
+      set(default_install lib/gui)
+
+    endif()
+     
+    if (ign_add_plugin_SCOPE)
+      add_library(${PLUGIN_NAME} ${ign_add_plugin_SCOPE} ${PLUGIN_PATH})
+    else()
+      add_library(${PLUGIN_NAME} SHARED ${PLUGIN_PATH})
+    endif()
+
+    set_property(TARGET ${PLUGIN_NAME} PROPERTY CXX_STANDARD 17)
+
+    target_link_libraries(${PLUGIN_NAME}
+      PUBLIC
+        ${ign_add_plugin_COMMON_PUBLIC_LIBRARIES}
+      PRIVATE
+        ${ign_add_plugin_COMMON_PRIVATE_LINK_LIBRARIES}
+    )
+    target_include_directories(${PLUGIN_NAME}
+      PUBLIC
+        ${ign_add_plugin_COMMON_PUBLIC_DIRECTORIES}
+      PRIVATE
+        ${ign_add_plugin_COMMON_PRIVATE_LINK_DIRECTORIES}
+    )    
+    if(ign_add_plugin_GUI)
+      target_link_libraries(${PLUGIN_NAME}
+        PRIVATE
+          ignition-gui${IGN_GUI_VER}::ignition-gui${IGN_GUI_VER}
+      )
+    endif()
+    if(ign_add_plugin_INSTALL_DESTINATION)
+      set(default_install ${ign_add_plugin_INSTALL_DESTINATION})
+    endif()
+    if(ign_add_plugin_GUI)
+      install(
+        TARGETS ${PLUGIN_NAME}
+        DESTINATION ${default_install})
+      list(APPEND gui_plugins_path ${default_install})
+    else()
+      install(
+        TARGETS ${PLUGIN_NAME}
+        DESTINATION ${default_install})
+      list(APPEND plugins_path ${default_install})
+    endif()  
+
+
+
+  endforeach()
+
+endmacro()
+
+################################################3
+#ign_export_plugin(path_to_install_destination)
+#Exports the provided installation path of plugin
+
+macro(ign_export_plugin path_to_install_destination)
+
+  list(APPEND plugins_path ${ign_export_plugin_INSTALL_DESTINATION})
+
+endmacro()  
+
+#####################################################
+#ign_add_executables(path_to_executables)
+#Installs all the executables inside the folder using various arguments from the user
+#Other than common link_libraries and directories,you can simply add specific executable dependencies one by one after this macro
+#The deafult file type is .cc for executables,you can add other file types using EXECUTABLES_EXTENSION argument
+
+macro(ign_add_executables path_to_executables)
+
+  set(oneValueArgs INSTALL_DESTINATION)
+  set(multiValueArgs COMMON_PUBLIC_LIBRARIES COMMON_PRIVATE_LINK_LIBRARIES COMMON_PUBLIC_DIRECTORIES COMMON_PRIVATE_LINK_DIRECTORIES EXECUTABLE_EXTENSION)
+
+  _ign_cmake_parse_arguments(ign_add_executable "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  if(NOT ign_add_executable_EXECUTABLE_EXTENSION)
+    list(APPEND ign_add_executable_EXECUTABLE_EXTENSION ".cc")
```

Comment:
Please don't do this. Users may have different coding standards within their org we should not enforce our standards on them.

Reply:
This is just a default extension, and it is mentioned in the comments. Any user can easily change it.